packages/ra-ui-materialui/src/input/CheckboxGroupInput.tsx

```diff
@@ -138,13 +138,19 @@ const CheckboxGroupInput: FunctionComponent<CheckboxGroupInputProps> = props =>
     const handleCheck = useCallback(
         (event, isChecked) => {
             let newValue;
-            try {
-                // try to convert string value to number, e.g. '123'
-                newValue = JSON.parse(event.target.value);
-            } catch (e) {
-                // impossible to convert value, e.g. 'abc'
+
+            if (value.every(item => typeof item !== 'number')) {
```

Comment:
no, you should iterate on choices, not value (which isn't an array by the way).

Also, the condition ('every item is not of tyype number') is harder to read than the opposite ('every item is a number'), so prefer the latter.